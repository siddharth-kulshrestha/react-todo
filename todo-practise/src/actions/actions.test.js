import { createItem, addMultipleItems, updateRating } from './index';

describe(' Testing action creators ', () => {
    it(' createItem ', () => {
        const result = createItem({content: 'test item content', rating: 50});
        console.log('Result: ');
        console.log(result);
        expect(result).toEqual({ type: 'CREATE_ITEM', payload: { content: 'test item content', rating: 50 } });
    });

    it(' createItem -- Snapshot tests ', () => { // This shows snapshot tests are not only limited to react. Its a feature of jest which compares JS objects.
        const result = createItem({content: 'test item content', rating: 50});
        expect(result).toMatchSnapshot();
    });
});
