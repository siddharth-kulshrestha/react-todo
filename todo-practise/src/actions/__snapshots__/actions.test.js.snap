// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Testing action creators   createItem -- Snapshot tests  1`] = `
Object {
  "payload": Object {
    "content": "test item content",
    "rating": 50,
  },
  "type": "CREATE_ITEM",
}
`;
